sentry-kotlin-multiplatform/src/appleMain/kotlin/io/sentry/kotlin/multiplatform/SentryScope.kt

```diff
@@ -0,0 +1,164 @@
+        }
+        get() {
+            val levelMap = scope?.serialize()?.get("level")
+            if (levelMap != null) {
+                val levelString = (levelMap as String).uppercase()
+                return SentryLevel.valueOf(levelString)
+            }
+            return null
+        }
+
+    actual override var user: SentryUser?
+        set(value) {
+            value?.onFieldChanged = {
+                scope?.setUser(value?.toCocoaSentryUser())
+            }
+        }
+        get() {
+            val userMap = scope?.serialize()?.get("user") as Map<String, String>?
+            if (userMap != null) {
+                val sentryUser = SentryUser()
+                sentryUser.email = userMap["email"] as String
+                sentryUser.username = userMap["username"] as String
+                sentryUser.id = userMap["id"] as String
+                sentryUser.ipAddress = userMap["ip_address"] as String
+                return sentryUser
+            }
+            return null
+        }
+
+    /**
+     * Initializes this KMP Scope with the Cocoa Scope
+     *
+     * @param cocoaScope: The Cocoa SDK scope
+     */
+    internal fun initWithCocoaScope(cocoaScope: CocoaScope) {
+        this.scope = cocoaScope
+    }
+
+    /**
+     * This initializes the SentryScope wrapper with the Cocoa scope and invokes the callback
+     * on this KMP scope which in turn modifies the Cocoa scope
+     */
+    internal fun scopeConfiguration(kmpScopeCallback: SentryScopeCallback): (CocoaScope?) -> Unit {
+        return {
+            initWithCocoaScope(it!!)
+            kmpScopeCallback.run(this)
+        }
+    }
+
+    actual override fun getContexts(): MutableMap<String, Any>? {
+        if (scope?.serialize()?.get("context") != null) {
+            val dict = scope?.serialize()?.get("context") as NSDictionary
+            val keys = dict.allKeys
+            val map: MutableMap<String, Any> = HashMap()
+            for (key in keys) {
+                dict.objectForKey(key)?.let { map.put(key as String, it) }
+            }
+            return map
+        }
+        return HashMap()
+    }
+
+    actual override fun getTags(): MutableMap<String, String>? {
+        if (scope?.serialize()?.get("tags") != null) {
+            val dict = scope?.serialize()?.get("tags") as NSDictionary
+            val keys = dict.allKeys
+            val map: MutableMap<String, String> = HashMap()
+            for (key in keys) {
+                map.put(key as String, dict.objectForKey(key) as String)
+            }
+            return map
+        }
+        return HashMap()
+    }
+
+    actual override fun addBreadcrumb(breadcrumb: SentryBreadcrumb) {
+        scope?.addBreadcrumb(breadcrumb.toCocoaBreadcrumb())
+    }
+
+    actual override fun clearBreadcrumbs() {
+        scope?.clearBreadcrumbs()
+    }
+
+    private fun setContextForAnyValue(key: String, value: Any) {
+        val map = HashMap<Any?, Any>()
+        map.put("value", value)
```

Comment:
Only JSON primitive types should be wrapped in the `value` object.
eg.
`{ "key": "myValue" }` would not work because the `value` should be an object, in this case we wrap it.
`{ "key": { "value": "myValue" } }`

Reply:
We just need to be sure that not every type here is wrapped into the `value` object.